aws/request/request_1_8_test.go

```diff
@@ -84,3 +84,77 @@ func TestRequest_FollowPUTRedirects(t *testing.T) {
 		t.Errorf("expect %d endpoint hits, got %d", e, a)
 	}
 }
+
+func TestNewRequest_JoinEndpointWithOperationPathQuery(t *testing.T) {
+	cases := map[string]struct {
+		HTTPPath    string
+		Endpoint    *string
+		ExpectQuery string
+		ExpectPath  string
+	}{
+		"no op HTTP Path": {
+			HTTPPath:    "",
+			Endpoint:    aws.String("https://foo.bar.aws/foo?bar=Baz"),
+			ExpectPath:  "/foo",
+			ExpectQuery: "bar=Baz",
+		},
+		"no trailing slash": {
+			HTTPPath:    "/",
+			Endpoint:    aws.String("https://foo.bar.aws"),
+			ExpectPath:  "/",
+			ExpectQuery: "",
+		},
+		"set query": {
+			HTTPPath:    "/?Foo=bar",
+			Endpoint:    aws.String("https://foo.bar.aws"),
+			ExpectPath:  "/",
+			ExpectQuery: "Foo=bar",
+		},
+		"squash query": {
+			HTTPPath:    "/?Foo=bar",
+			Endpoint:    aws.String("https://foo.bar.aws/?bar=Foo"),
+			ExpectPath:  "/",
+			ExpectQuery: "Foo=bar",
+		},
```

Comment:
Oh didn't know this is the standard behavior.

Reply:
This preserves the current behavior where query strings defined in an endpoint are squashed by any query string defined in the API operation's modeled `requestUri` property.